### ISSUE TITLE
DBAL-999: Get a Sql Server error on Order By - Symfony2

### ISSUE THREAD

**Issue by doctrinebot** [doctrine/dbal#2247]
Jira issue originally created by user neufman:

Using Symfony with Sql Server and from what I've read, it seems that the connection to the database is not stable.

As soon as I use the orderBy method I get an error :

Here's an example :

```php
  $qStores =
        $this->getManager()
             ->createQueryBuilder()
             ->select('rpdv')
             ->from('MainBundle:PointDeVenteReference', 'rpdv')
             ->andWhere( 'rpdv.partenaireClient = :id_partner ' )
                 ->setParameter( 'id_partner', $this->getUser()->getPartenaire()->getIdPartenaire() )
             ->orderBy( 'rpdv.idPointDeVenteReference' , 'DESC' )
             ->setFirstResult( 0 )
             ->setMaxResults( 30 );
```

And the error :

{quote}
An exception has been thrown during the rendering of a template ("An exception occurred while executing
'SELECT DISTINCT TOP 30 id*point_de_vente*reference0 
 FROM ( SELECT p0*.id_point_de_vente_reference AS id_point_de_vente*reference0, 
               p0_.reference AS reference1, 
               p0*.date_derniere_modification AS date_derniere*modification2, 
               p0_.blocage AS blocage3 
        FROM point*de_vente_reference p0* 
        WHERE p0*.id_partenaire*client = ? 
        ORDER BY p0*.id_point_de_vente_reference DESC ) dctrn*result 
 ORDER BY id*point_de_vente*reference0 DESC' 
 with params [2829]:SQLSTATE[42000]: 
 [Microsoft][SQL Server Native Client 11.0][SQL Server]
 The ORDER BY clause is invalid in views, inline functions, derived tables, subqueries, and common table expressions, 
 unless TOP, OFFSET or FOR XML is also specified.") in MainBundle:Default:store/list.html.twig at line 79.
I tried to change the class SQLServerPlatform with corrections found on the net, without success.
{quote}

Do you have any idea?

Thx !

**Comment by deeky666** [comment]
@zeroedin-bill couldyou please investigate whether this might already be fixed and close this issue eventually? Thanks.

**Comment by zeroedin-bill** [comment]
Fixed in orm 2.5, definitely fixed in dbal 2.5.3.
